examples/pipelines/commoncrawl/components/extract_free_to_use_images/src/main.py

```diff
@@ -0,0 +1,94 @@
+import re
+import logging
+
+import pandas as pd
+from bs4 import BeautifulSoup
+from typing import List
+
+from fondant.component import PandasTransformComponent
+
+from utils.license_utils import get_license_type, get_license_location
+from utils.image_utils import get_images_from_soup, get_unique_images
+
+logger = logging.getLogger(__name__)
+
+
+def get_image_info_from_webpage(webpage_url: str, webpage_html: str) -> List[str]:
+    """Extracts image urls and license metadata from the parsed html code.
+    Args:
+        webpage_url: The url of the webpage.
+        webpage_html: The html content of the webpage.
+    Returns:
+        A list of image urls and license metadata.
+    """
+
+    soup = BeautifulSoup(webpage_html, "html.parser")
```

Comment:
perhaps this should go in the `setup()` method? Seems like you're now initializing this on every function call

Reply:
doesn't look like `BeautifulSoup` has a a parser function without the constructor so I'm leaving it as it is